### PR TITLE
New: Integration Window will be the title of the integration

### PR DESCRIPTION
**Pull request by jeremypress**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @jeremypress has signed the CLA. Thanks for the pull request!

**Comment by boxcla** [comment]
# Webpack Bundle Size Github Bot

| Bundle        | Diff | Percentage Diff | Before | After |
| ------------- |:----:|:---------------:|:------:|:-----:|
| Total | 0 B | 0.00% | 0 B | 0 B |

Compared against previously merged PR #613

**Powered by [webpack-bundle-size-github-bot](https://github.com/wenboyu2/webpack-bundle-size-github-bot/)**

File: src/components/ContentOpenWith/OpenWithButton.js [hunk 1]
```diff
@@ -20,7 +19,7 @@ type Props = {
     error: ?any,
     icon?: string,
     isLoading: boolean,
-    onClick: ?Function,
+    onClick: Function,
```
[the remark this answers is not in the document]
Comment (jeremypress) [reply src/components/ContentOpenWith/OpenWithButton.js]: with the introduction of onIntegrationClick, we will always pass that in as our on click handler, so there's no need to default something that should always have a value.

File: src/components/ContentOpenWith/ContentOpenWith.js [hunk 2]
```diff
@@ -241,14 +241,16 @@ class ContentOpenWith extends PureComponent<Props, State> {
...
+    onIntegrationClick = (displayIntegration: Integration): void => {
+        const { appIntegrationId, displayName } = displayIntegration;
         const { fileId }: Props = this.props;
         // window.open() is immediately invoked to avoid popup-blockers
         // The name is included to be the target of a form if the integration is a POST integration
         this.integrationWindow = this.window.open(
             '',
             `OpenWithIntegration-${appIntegrationId}`,
         );
+        this.integrationWindow.document.title = `${displayName}`;
```
[the remark this answers is not in the document]
Comment (jeremypress) [reply src/components/ContentOpenWith/ContentOpenWith.js]: done.

File: src/components/ContentOpenWith/ContentOpenWith.js [hunk 3]
```diff
@@ -241,14 +241,16 @@ class ContentOpenWith extends PureComponent<Props, State> {
      * @private
      * @return {void}
      */
-    onIntegrationClick = (appIntegrationId: string): void => {
+    onIntegrationClick = (displayIntegration: Integration): void => {
```
Comment (jstoffan) [inline src/components/ContentOpenWith/ContentOpenWith.js]: You can destructure this in the function signature, if you want:

`onIntegrationClick = ({ appIntegrationId, displayName }: Integration): void => {`

File: src/components/ContentOpenWith/OpenWithButton.js [hunk 4]
```diff
@@ -72,7 +70,7 @@ const OpenWithButton = ({
         >
             <Button
                 isDisabled={isDisabled}
-                onClick={onClick ? () => onClick(id) : noop}
+                onClick={onClick(displayIntegration)}
```
Comment (jstoffan) [inline src/components/ContentOpenWith/OpenWithButton.js]: This changed from a function declaration to a direct invocation. Was that intentional?

Comment (jeremypress) [reply src/components/ContentOpenWith/OpenWithButton.js]: Fixed

**Pull request merged**